client/src/context/MouseEvents/MouseEvents.tsx

```diff
@@ -150,7 +156,7 @@ export const MouseEventsProvider = ({
         w: viewBox.w - dw,
         h: viewBox.h - dh,
       };
-      const scale = viewBox.w / w;
+      const scale = newViewBox.w / w;
```

Comment:
Add tests for this onScroll feature, also perhaps consider moving this to another place. Perhaps the canvas?